src/number_theory/numbers.rs

```diff
@@ -0,0 +1,32 @@
+//! Number Trait - For more convenient trait bounding
+use cgmath::{BaseFloat, BaseNum};
+use num::{Bounded, CheckedMul, FromPrimitive, ToPrimitive};
+use rand::distributions::uniform::SampleUniform;
+
+pub trait Integers:
+    BaseNum + ToPrimitive + FromPrimitive + SampleUniform + Bounded + Send + Sync + Ord + CheckedMul
+{
+}
+
+impl<T> Integers for T where
+    T: BaseNum
+        + ToPrimitive
+        + FromPrimitive
+        + SampleUniform
```

Comment:
Not all of these traits are needed to cover the set of all integers. However, there are some desired additional functions for other purposes. Such as the ability to generate random integers along a uniform sample number line. The ability to enforce orders and sort. Finally, checked multiplication which is desirable for error checking. However, to actually cover the basics of what an Integer is, BaseNum, Bounded, Send, Sync should be sufficient.

Reply:
The additional traits belong here due to the dependency tree existent between the different modules being worked on (to be committed to separate branches when ready).